### PR TITLE
Folly/2022.10.03

### PR DESCRIPTION
**Pull request by Kidsunbo**
Specify library name and version:  **Folly/2022.10.03**

Update folly to 2022.10.03 and add one patch

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/12988

### CODE REVIEW

**Comment by uilianries** [comment]
@Kidsunbo Thank you for your contribution, please, don't forget to sign our CLA on https://github.com/conan-io/conan-center-index/pull/13290#issuecomment-1266344222. The CI bot does not allow to merge PRs which are not signed.

**Review by prince-chrismc — COMMENTED** [review]
Since the patch is to provide conan deps you should also change the mode, the upstream projects all provide CONFIG files, modules are a different and I dont think any of these are on 

https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules

**Comment by Kidsunbo** [comment]
> [request] Folly/v2022.09.12.00 #12988

Sorry I haven't used this before. I just did it and don't know if I've done it correctly.

**Comment by Kidsunbo** [comment]
> Since the patch is to provide conan deps you should also change the mode, the upstream projects all provide CONFIG files, modules are a different and I dont think any of these are on
> 
> https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules

I don't know why but when I accept the change from MODULE to CONFIG, it fails to finish the CMake configuration on my machine. So I remove the keyword and make it use whatever it can use.

**Comment by uilianries** [comment]
@Kidsunbo Need to check which generator is being used. The `cmake_find_package` generates only `Findxxx.cmake` files, and `cmake_find_package_multi` generates `xxx-config.cmake`.

**Comment by prince-chrismc** [comment]
Thats a really interesting point, it might have a cascade problem with the other version 🤔 
I would play it safe and not change the generators lets just try adding the version before we play with it too much... with the new generators the behavior will change too 😱

**Comment by dubvulture** [comment]
Why the upgrade to glog/0.6.0 and fmt/9.1.0?
The folly manifest files report [0.5.0](https://github.com/facebook/folly/blob/v2022.10.03.00/build/fbcode_builder/manifests/glog) and [8.0.1](https://github.com/facebook/folly/blob/v2022.10.03.00/build/fbcode_builder/manifests/fmt), if I interpreted them correctly.

**Comment by uilianries** [comment]
@dubvulture We try to avoid dependencies conflicts. If folly only work with those versions, it's fine keep them, otherwise, update them.

**Comment by Kidsunbo** [comment]
> @Kidsunbo Need to check which generator is being used. The `cmake_find_package` generates only `Findxxx.cmake` files, and `cmake_find_package_multi` generates `xxx-config.cmake`.

The generators seems to be `cmake` and `cmake_find_package`. So I think the Module mode should be used.

**Review by prince-chrismc — COMMENTED** [review]
Can you please fix this, linter is flagging some bad code should be `is_msvc(self)` https://github.com/conan-io/conan-center-index/pull/13290/files#diff-c76b59bc54e55f5a4d20622555a9653f5345015c0c85716862cdab5139eb4dc6R162

it's line 162

otherwise it's a green check!

**Comment by Kidsunbo** [comment]
> https://github.com/conan-io/conan-center-index/pull/13290/files#diff-c76b59bc54e55f5a4d20622555a9653f5345015c0c85716862cdab5139eb4dc6R162

Yeah I have fix this by replacing `is_msvc` to `is_msvc(self)`. And also fix one error reported by lint caused by V2 migration. There is still one error which is also related to V2 migration, I have tried but it seems changed a lot and the generator needs to be changed as well and I think I should be careful with this. So I left the last error, and hope you guys could help me fix it if needed. Thanks.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
I think its more important to get the version in then worry about changing the recipes. this is a piecemeal one

**Comment by prince-chrismc** [comment]
The PR to add you is in the queue, I am not sure why it's stuck again but I asked the infra team for help 🤞 it shouldn't be long

File: recipes/folly/all/conanfile.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from conan.tools.scm import Version
 from conan.tools import files
 from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/folly/all/conanfile.py:5]:
```suggestion
from conan import ConanFile
from conans import CMake, tools
```

**Review by uilianries — COMMENTED** [review]
Linter fix.

**Review by prince-chrismc — DISMISSED** [review]

**Comment by kexianda** [comment]
@uilianries  @prince-chrismc 

It is a bit annoying to create patch files. 
In this case,  0023-find-packages.patch very similar with 0016-find-packages.patch.  
Most of them are just string replacement.  
I am wondering whether it is OK to provide an other regex expression api like this?
```python
def build(self):
      # ...
       if Version(self.version) >= "2022.01010.03"
            re = ['s/DoubleConversion/double-conversion/g', 's/Gflags/gflags/g', ....]
            tools.patch_regex(fp='CMake/folly-deps.cmake'  re)
        cmake = self._configure_cmake()
        cmake.build()

```

**Comment by prince-chrismc** [comment]
https://github.com/conan-io/conan-center-index/blob/master/docs/policy_patching.md#patches-format-and-conventions

Yes it's allowed - but they can have unintended side effects which are difficult to review.

It's very easy to make patches with the codespaces from github, you just right click export diff, if you are interested in other solutions.

**Comment by kexianda** [comment]
> https://github.com/conan-io/conan-center-index/blob/master/docs/policy_patching.md#patches-format-and-conventions

Thank you for your reply.

**Comment by michaelaeriksen** [comment]
Any plans to wrap this up? Seems like a waste of good work not to finish this port.

**Comment by Kidsunbo** [comment]
> Any plans to wrap this up? Seems like a waste of good work not to finish this port.

There is a PR to bump up folly to a newer version than this one. So I think we should wait that PR to be merged.

**Pull request closed without merging**